webapp/client/src/index.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import LoginPage from './pages/LoginPage';
+
+const componentNameMap = {
+  'LoginPage': LoginPage
+}
+
+document.querySelectorAll('[data-is-component=yes]').forEach( (element) => {
+  const name = element.dataset.componentName;
+  const Component = componentNameMap[name];
+  if (Component !== undefined) {
+    const props = element.dataset.propsJson ? JSON.parse(element.dataset.propsJson) : {}
+    ReactDOM.render(
+      <React.StrictMode>
+        <Component {...props} />
+      </React.StrictMode>,
+      element
+    );
+  } else {
+    console.log(`No such component "${name}"`)
```

Comment:
This is probably a place where we want to think about error handling

Reply:
Do you have anything specific in mind?

Reply:
I was just thinking that we might want to be informed about missing components. But that depends a lot on if and how we want to log client errors

Reply:
Well, as it stands, client-side errors are logged in the users' browser consoles. We could think about adding infrastructure to report back errors to our servers, but that's a project in itself ([which does have a ticket](https://steuerlotse.atlassian.net/browse/STL-1362?atlOrigin=eyJpIjoiNWUwYzFlYzc4ODJjNDIwZTgwZTNhNDgwN2Q4M2ZlNzQiLCJwIjoiaiJ9)) and probably needs some consideration of privacy concerns as well.